js/publisher.js

```diff
@@ -0,0 +1,173 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const Immutable = require('immutable')
+const sinon = require('sinon')
+
+const Publisher = require('../abs/publisher')
+const {request} = require('../lib/util/request')
+const settings = require('../browser-laptop/js/constants/settings')
+const ledgerState = require('../browser-laptop/app/common/state/ledgerState')
+const responses = require('../lib/data/responses.json')
+
+const defaultAppState = Immutable.fromJS({
+  cache: {
+    ledgerVideos: {}
+  },
+  ledger: {
+    about: {}
+  },
+  about: {
+    preferences: {}
+  },
+  tabs: [],
+  windows: [{
+    focused: true,
+    windowId: -1
+  }]
+})
+
+class JS extends Publisher {
+  constructor () {
+    super()
+    this.ledger = null
+    this.timeStamp = 1525688397657
+    this.state = defaultAppState
+    this.mediaType = null
+    this.mediaMinimum = null
+  }
+
+  get settingsP () {
```

Comment:
I see that this function is used in other tests as well. Can we pull it in shared?